core/htmlparser/nameFilterRule.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license Copyright (c) 2003-2018, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @fileOverview Defines the "virtual" {@link CKEDITOR.htmlParser.nameFilterRule} class
+ * that contains the definition of rule for filtering element names or attribute names. This file is for
+ * documentation purposes only.
+ */
+
+/**
+ * Abstract class describing the definition of {@link CKEDITOR.htmlParser.filterRulesDefinition} `elementNames` and `attributesNames` filtering rules.
+ *
+ * ```javascript
+ *  var rule = [ /^div$/, 'p' ];
+ * ```
+ *
+ * @class CKEDITOR.htmlParser.nameFilterRule
```

Comment:
I'm wondering about the name of the class. It's used for elements's transformations, so maybe `CKEDITOR.htmlParse.transformRule`?

Reply:
Every member of `CKEDITOR.htmlParser.filterRulesDefinition` class can used for some kind of transformation. Although only two uses this abstract class. `transformRule` sounds a bit misleading to me.
What do you think about `nameTransformRule`? These rules alters element names or attributes, but they don't touch values/content so this name would be a bit more precise. WDYT @Comandeer?